composer/trainer/checkpoint.py

```diff
@@ -427,11 +462,17 @@ def save_checkpoint(self, state: State, seed: int, device: Device) -> None:
                 with open(composer_states_filepath, 'xb') as f:
                     torch.save(state_dict, f)
 
-            checkpoint_archive_filepath = os.path.join(self.checkpoint_folder, f'{tag}{self.file_extension}')
-            with tarfile.open(checkpoint_archive_filepath, self.write_mode) as tarball:
-                tarball.add(tmpdir, arcname="")  # add files flat to the tarball
-
-            log.info(f'Trainer checkpoint saved to {checkpoint_archive_filepath}')
+            checkpoint_filepath = os.path.join(self.checkpoint_folder, f'{tag}{self.file_extension}')
+            if not _is_pt_file(checkpoint_filepath):
+                with tarfile.open(checkpoint_filepath, self.write_mode) as tarball:
+                    # add files flat to the tarball with the specified compression
+                    tarball.add(tmpdir, arcname="")
```

Comment:
Can you set the `arcname` to be something like the checkpointing timestamp and see if that fixes the permission issues we've been seeing?

Reply:
I tried it didn't :/